### ISSUE TITLE
Eventlist: Month view shows events from past months from multi-day event

### ISSUE THREAD

**Issue by arno1979** [contao/core#8792]
An error similar to the one described in #8782 and fixed in https://github.com/contao/core/commit/03d2a8b08dad2c30601058d8549a322a103e36a8 appears when using an eventlist where `cal_format` is set to `month` (applies to the setting `day` as well.)
But as far as I can tell the source of the error lies somewhere completely different therefore I opened this new issue.

Steps to reproduce:
* create a multi-day event stretching from 2017-08-15 to 2017-11-22
* create a list module with `cal_format` set to `month` and the option `shortened_view` set to disabled
* include the list module on a page and load that page with the get parameter `?month=201709`

Expected result:
* show all dates from the month of September (2017-09-01 to 2017-09-30)

Actual result:
* all dates from the start of the event (2017-08-15) until the end of September are shown (see also screenshot)
<img width="644" alt="bildschirmfoto 2017-09-28 um 09 31 52" src="https://user-images.githubusercontent.com/1838113/30956895-ec6c9a8c-a438-11e7-9b11-cc84ca9b87f1.png">

I think I also found the source of the error:
In the function `addEvent` in `Events.php` events are added by using the `$intStart` variable on lines 213 - 218.
In a multi-day event this always seems to be the date of the first day of the event.
I experimented with using `$intBegin` instead by adding `$intStart = $intBegin;` above line 213 and got the desired result:
Only dates from September are shown.

I am not sure if you can simply replace `$intStart` with `$intBegin` because I have not checked the implications if the list module is configured differently than in the case described here.

**Comment by leofeyer** [comment]
We have changed the behavior because of #8194. Can you please test the following code:

```php
protected function addEvent($objEvents, $intStart, $intEnd, $strUrl, $intBegin, $intLimit, $intCalendar)
{
	/** @var \PageModel $objPage */
	global $objPage;

	$span = \Calendar::calculateSpan($intStart, $intEnd);

	// Adjust the start time of a multi-day event (see #8792)
	if (!$this->cal_noSpan && $span > 0)
	{
		$intStart = $intBegin;
	}

	$intDate = $intStart;
	// …
```

**Comment by arno1979** [comment]
I made a quick test and your code seems to work!

**Comment by leofeyer** [comment]
Fixed in c962d5caa4523de29a3494d3ab541ad704c6199d. Did you also make sure that the code does not alter any of the other modes?

**Comment by leofeyer** [comment]
I did some extensive testing and came to the conclusion that the code did not work. I have now fixed the issue in b509bbadfed78a6379e2cb2a150ef8f225f7d5fe.
